x-pack/test/rule_registry/spaces_only/tests/trial/get_summarized_alerts.ts

```diff
@@ -0,0 +1,278 @@
+      pluginStop$ = new ReplaySubject(1);
+
+      const ruleDataService = new RuleDataService({
+        getClusterClient,
+        logger,
+        kibanaVersion: '8.0.0',
+        isWriteEnabled: true,
+        isWriterCacheEnabled: false,
+        disabledRegistrationContexts: [] as string[],
+        pluginStop$,
+      });
+
+      // This initializes the service. This happens immediately after the creation
+      // of the RuleDataService in the setup phase of the Rule Registry plugin
+      ruleDataService.initializeService();
+
+      // This initializes the index and templates and returns the data client.
+      // This happens in each solution plugin before they can register lifecycle
+      // executors.
+      ruleDataClient = ruleDataService.initializeIndex({
+        feature: AlertConsumers.OBSERVABILITY,
+        registrationContext: 'observability.test.alerts',
+        dataset: Dataset.alerts,
+        componentTemplateRefs: [],
+        componentTemplates: [
+          {
+            name: 'mappings',
+            mappings: mappingFromFieldMap(
+              {
+                testObject: {
+                  type: 'object',
+                  required: false,
+                  array: false,
+                },
+              },
+              false
+            ),
+          },
+        ],
+      });
+    });
+
+    after(async () => {
+      cleanupRegistryIndices(getService, ruleDataClient);
+      pluginStop$.next();
+      pluginStop$.complete();
+    });
+
+    it('should return new, ongoing and recovered alerts', async () => {
+      const id = 'host-01';
+
+      // This creates the function that will wrap the solution's rule executor with the RuleRegistry lifecycle
+      const createLifecycleRuleExecutor = createLifecycleExecutor(logger, ruleDataClient);
+      const createGetSummarizedAlerts = createGetSummarizedAlertsFn({
+        ruleDataClient,
+        useNamespace: false,
+        isLifecycleAlert: true,
+      });
+
+      // This creates the executor that is passed to the Alerting framework.
+      const executor = createLifecycleRuleExecutor<
+        MockRuleParams,
+        { shouldTriggerAlert: boolean },
+        MockAlertState,
+        MockAlertContext,
+        MockAllowedActionGroups
+      >(async function (options) {
+        const { services, state: previousState } = options;
+        const { alertWithLifecycle } = services;
+
+        const triggerAlert = previousState.shouldTriggerAlert;
+
+        if (triggerAlert) {
+          alertWithLifecycle({
+            id,
+            fields: {
+              [ALERT_REASON]: 'Test alert is firing',
+            },
+          });
+        }
+
+        return Promise.resolve({ shouldTriggerAlert: triggerAlert });
+      });
+
+      const getSummarizedAlerts = createGetSummarizedAlerts();
+
+      // Create the options with the minimal amount of values to test the lifecycle executor
+      const options = {
```

Comment:
send-ahead; not happy with this server-side-y FTR style of test.  I just had to patch one due to some changes with the alert factory.  Issue opened: https://github.com/elastic/kibana/issues/144557

I don't think it's worth the time to fix this in this PR, since you already wrote it (I'll have to fix it when I merge my PR, but I already know how :-) ).  However we fix the one in the issue, will likely be the same as this.  It makes me grumpy, but not worth spending time to fix right now.

Reply:
🙈 I think this could be removed in a future PR when we're actually requesting alert summaries in the alerting task runner and then the FT would be able to test that whole flow. Since this is just a piece of the puzzle, creating and registering the function but not actually using it anywhere, I went with this test.

Reply:
ya :-).  I may well have suggested you take this route anyway, since there was already one that we should eventually fix.